components/proxy/src/main/java/com/hotels/styx/admin/handlers/ProviderRoutingHandler.java

```diff
@@ -67,16 +89,90 @@ private void refreshRoutes(ObjectStore<? extends StyxObjectRecord<? extends Styx
 
     private UrlPatternRouter buildRouter(ObjectStore<? extends StyxObjectRecord<? extends StyxService>> db) {
         UrlPatternRouter.Builder routeBuilder = new UrlPatternRouter.Builder(pathPrefix)
-                .get("", new ProviderListHandler(db));
+            .get("", new ProviderListHandler(db))
+            .get("objects", (request, context) -> {
+                return handleRequestForAllObjects(request, context, db);
+            })
+            .get("objects/:objectName", (request, context) -> {
+                String name = placeholders(context).get("objectName");
+                return handleRequestForOneObject(request, context, db, name);
+            });
 
         db.entrySet().forEach(entry -> {
-                String providerName = entry.getKey();
-                entry.getValue().getStyxService().adminInterfaceHandlers(pathPrefix + "/" + providerName)
-                        .forEach((relPath, handler) ->
-                            routeBuilder.get(providerName + "/" + relPath, new HttpStreamer(MEGABYTE, handler))
-                        );
+            String providerName = entry.getKey();
+            entry.getValue().getStyxService().adminInterfaceHandlers(pathPrefix + "/" + providerName)
+                .forEach((relPath, handler) ->
+                    routeBuilder.get(providerName + "/" + relPath, new HttpStreamer(MEGABYTE, handler))
+                );
         });
 
         return routeBuilder.build();
     }
+
+    private Eventual<HttpResponse> handleRequestForAllObjects(HttpRequest request, HttpInterceptor.Context context,
+                                                              ObjectStore<? extends StyxObjectRecord<? extends StyxService>> db) {
+        List<StyxObjectDefinition> objects = db.entrySet()
+                .stream()
+                .map(entry -> createObjectDef(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+        String output = serialise(objects);
```

Comment:
An opinionated comment, but I would inline the `objects` because it is only ever passed on to `serialise` method. ;-)

Reply:
Hm. In this case I find it flows better as it is, because the stream construction is (in my mind) quite different to the method call, and inlining just results in my eye bouncing around a little more than I'd like.
Ideally I'd have liked to be able to follow the .collect() with .serialise() (or some construction with ::serialise) but that's not available. Visually, I think this is the next best thing. But I'll change it (and the others) if you feel moderately strongly about it.